src/Modules/Guilds/Hooks/useProposalVotesOfVoter.ts

```diff
@@ -0,0 +1,44 @@
+import ERC20Guild from 'contracts/ERC20Guild.json';
+import { useContractRead } from 'wagmi';
+import { BigNumber } from 'ethers';
+import { useMemo } from 'react';
+import { WagmiUseContractReadResponse } from 'Modules/Guilds/Hooks/types';
+
+export interface UseProposalVotesOfVoterReturn {
+  action: string;
+  votingPower: BigNumber;
+}
+
+const useProposalVotesOfVoter = (
+  guildAddress: string,
+  proposalId: string,
+  userAddress: string
+): WagmiUseContractReadResponse<UseProposalVotesOfVoterReturn> => {
+  const { data, ...rest } = useContractRead({
+    enabled: !!guildAddress && !!proposalId && !!userAddress,
+    addressOrName: guildAddress,
+    contractInterface: ERC20Guild.abi,
+    functionName: 'getProposalVotesOfVoter',
+    args: [proposalId, userAddress],
+    watch: true,
+  });
+
+  const parsedData = useMemo<UseProposalVotesOfVoterReturn>(() => {
+    if (!data?.votingPower || !data?.action)
+      return { action: null, votingPower: null };
+    if (BigNumber.from(data.votingPower || 0).gt(0)) {

```

Comment:
Do we need to make a check for `data.action` here as well? Could it also explode here without a `data?.votingPower`?

Reply:
L27 is checking that so can't reach 29 with nullish/undefined values.